includes/blocks/class-sensei-course-navigation-block.php

```diff
@@ -0,0 +1,285 @@
+
+	/**
+	 * Render Course Navigation View block.
+	 *
+	 * @access private
+	 *
+	 * @return string Block HTML.
+	 */
+	public function render_course_navigation() {
+
+		$course_id = Sensei_Utils::get_current_course();
+
+		if ( ! $course_id ) {
+			return '';
+		}
+
+		$structure = Sensei_Course_Structure::instance( $course_id )->get();
+
+		$modules_html = implode(
+			'',
+			array_map(
+				function( $item ) use ( $course_id ) {
+					if ( 'module' === $item['type'] ) {
+						return $this->render_module( $item, $course_id );
+					}
+					return '';
+				},
+				$structure
+			)
+		);
+
+		$lessons_html = implode(
+			'',
+			array_map(
+				function( $item ) use ( $course_id ) {
+					if ( 'lesson' === $item['type'] ) {
+						return $this->render_lesson( $item );
+					}
+					return '';
+				},
+				$structure
+			)
+		);
+
+		return '<div class="sensei-lms-course-navigation">
+			<div class="sensei-lms-course-navigation__modules">
+				' . $modules_html . '
+			</div>
+			<div class="sensei-lms-course-navigation__lessons">
+				' . $lessons_html . '
+			</div>
+			' . $this->render_svg_icon_library() . '
+		</div>';
+	}
+
+
+	/**
+	 * Build module block HTML.
+	 *
+	 * @param array $module    Module data.
+	 * @param int   $course_id The course id.
+	 *
+	 * @return string Module HTML
+	 */
+	public function render_module( $module, $course_id ) {
+
+		$module_id  = $module['id'];
+		$title      = esc_html( $module['title'] );
+		$lessons    = $module['lessons'];
+		$module_url = add_query_arg( 'course_id', $course_id, get_term_link( $module_id, 'module' ) );
+
+		$lessons_html = implode(
+			'',
+			array_map(
+				function( $lesson ) {
+					return $this->render_lesson( $lesson );
+				},
+				$lessons
+			)
+		);
+
+		$has_current_lesson = count(
+			array_filter(
+				$lessons,
+				function( $lesson ) {
+					return Sensei_Utils::get_current_lesson() === $lesson['id'];
+				}
+			)
+		);
+		$is_current_module  = get_the_ID() === $module_id || $has_current_lesson;
+
+		$lesson_count = count( $lessons );
+		$quiz_count   = count(
+			array_filter(
+				$lessons,
+				function( $lesson ) {
+					return Sensei_Lesson::lesson_quiz_has_questions( $lesson['id'] );
+				}
+			)
+		);
+
+		// Translators: placeholder is number of lessons.
+		$summary_lessons = _n( '%d lesson', '%d lessons', $lesson_count, 'sensei-lms' );
+		// Translators: placeholder is number of quizzes.
+		$summary_quizzes = _n( '%d quiz', '%d quizzes', $quiz_count, 'sensei-lms' );
+		$summary         = sprintf( $summary_lessons . ', ' . $summary_quizzes, $lesson_count, $quiz_count );
+
+		$classes   = [ 'sensei-lms-course-navigation-module sensei-collapsible' ];
+		$collapsed = '';
+		if ( ! $is_current_module ) {
+			$collapsed = 'collapsed';
+		}
+
+		$collapse_toggle = '';
+		if ( ! empty( $this->settings['collapsibleModules'] ) ) {
+			$collapse_toggle = '<button type="button" class="sensei-lms-course-navigation__arrow sensei-collapsible__toggle ' . $collapsed . '">
```

Comment:
TwentyTwenty One is adding a background to the button. We could add a `background: transparent` to the arrow.

Reply:
Should be fixed by 6b6b77e